src/main/java/com/uid2/admin/store/MultiScopeStoreWriter.java

```diff
@@ -49,17 +61,33 @@ private Map<Integer, T> getCurrentState(Collection<Integer> siteIds) throws Exce
         return currentState;
     }
 
-    private void write(Map<Integer, T> desiredState, Collection<Integer> sitesToWrite, JsonObject extraMeta) throws Exception {
-        for (Integer addedSite : sitesToWrite) {
-            factory.getWriter(addedSite).upload(desiredState.get(addedSite), extraMeta);
+    public void uploadEncrypted(Map<Integer, T> desiredState, JsonObject extraMeta) throws Exception {
+        EncryptedStoreFactory<T> encryptedFactory = (EncryptedStoreFactory<T>) factory;
+        for (Map.Entry<Integer, T> entry : desiredState.entrySet()) {
+            Integer siteId = entry.getKey();
+            if (siteId != null) {
```

Comment:
When would siteId be null? What should be done in this case?

Reply:
This siteId would never be null, as ensured by getPublicSitesMap in PublicSiteUtil. I can remove this check if thats better